apollo-compiler/src/main/kotlin/com/apollostack/compiler/ir/Field.kt

```diff
@@ -1,31 +1,46 @@
 package com.apollostack.compiler.ir
 
+import com.cesarferreira.pluralize.singularize
 import com.squareup.javapoet.MethodSpec
 import com.squareup.javapoet.TypeName
 import javax.lang.model.element.Modifier
 
 data class Field(
-  val responseName: String,
-  val fieldName: String,
-  val type: String,
-  val fields: List<Field>?
-) {
-
-  fun toTypeName(prefix: String? = null) = "${prefix?.capitalize() ?: ""}${type.normalizeTypeName()}"
-
-  fun toMethodSpec(returnTypeOverride: String? = null): MethodSpec {
-    val typeName = returnTypeOverride
-      ?.let { if (type.startsWith('[') && type.endsWith(']')) "[${it.normalizeTypeName()}]" else it.normalizeTypeName() }
-      ?.toTypeName()
-      ?: type.toTypeName()
-
+    val responseName: String,
+    val fieldName: String,
+    val type: String,
+    val fields: List<Field>?) {
+  fun toMethodSpec(): MethodSpec {
     return MethodSpec.methodBuilder(responseName)
-      .returns(typeName)
-      .addModifiers(Modifier.PUBLIC, Modifier.ABSTRACT)
-      .build()
+        .returns(toTypeName(methodResponsType()))
+        .addModifiers(Modifier.PUBLIC, Modifier.ABSTRACT)
+        .build()
   }
 
-  private fun String.toTypeName(): TypeName = GraphQlType.resolveByName(this).toJavaTypeName()
+  private fun toTypeName(responseType: String): TypeName =
+      GraphQlType.resolveByName(responseType).toJavaTypeName()
+
+  private fun methodResponsType(): String {
+    // TODO: Do we also need to handle nested lists?
```

Comment:
I thought our GraphQlType class supports nested lists, I remember I even added test case for that, or I missing smth?